### DISCUSSION TITLE
Help for multipage rendering

### DISCUSSION THREAD

**Discussion by pridar2k7** [pypdfium2-team/pypdfium2#245] [Q&A]
I am coming up with a code for rendering multipages(parallel) but the PdfDocument was created using the pdf bytes. Can you help me figure out an option to do it? This is how i am loading the document.

```python
def load_pdf_file(pdf_content: bytes, filename: str) -> PdfDocument:
     pdf = pdfium.PdfDocument(pdf_content)
     return pdf

page_indices = [i for i in range(len(pdf_document))]

renderer = pdf_document.render(
    pdfium.PdfBitmap.to_pil,
    page_indices=page_indices,
)
```

**Comment by mara004** [comment]
You had better use the single-page renderer and maybe implement parallelization on caller side.
The API design and code of `PdfDocument.render()` was bad, unfortunately. 🫤
I'm currently during a major rewrite, which will either remove or fundamentally change this API.
Sorry for the inconveniences.

**Reply by pridar2k7** [reply]
Got it! Thank you for your response! Does it make sense to do the single page render as below? Or do you recommend use of a different method? (the load_pdf will be done as above)   

```python
page = pdf_document[page_index]
pil_image = page.render(
    scale=dpi/72,
    rotation=0,
).to_pil()
```

**Reply by mara004** [reply]
Looks good! Only one nit, rotation defaults to 0, so you needn't pass it.
